cmake/custom/main.cmake

```diff
@@ -5,12 +5,13 @@ configure_file (
     "${PROJECT_BINARY_DIR}/config.h"
     )
 
-include_directories(${PROJECT_BINARY_DIR})
-
-include_directories(${PROJECT_BINARY_DIR}/external/include)
-link_directories(${PROJECT_BINARY_DIR}/external/lib)
+install(
+    FILES ${CMAKE_BINARY_DIR}/config.h
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}/src
```

Comment:
I think the destination of the `config.h` header should be just `${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}`